valhalla/meili/routing.h

```diff
@@ -23,107 +23,139 @@ namespace meili {
 constexpr uint16_t kInvalidDestination = std::numeric_limits<uint16_t>::max();
 
 /**
- * Labels used to mark edges in the path. Includes predecessor and cost
- * information needed to construct and recover shortest paths.
- * TODO simplify by inheriting from sif::EdgeLabel
+ * Label used to mark edges within the map-matching routing algorithm.
+ * Derived from EdgeLabel, this adds information required by map-matching
+ * which uses a one to many routing algorithm.
  */
-struct Label
-{
-  Label() = delete;
-
-  Label(const baldr::GraphId& the_nodeid, const baldr::GraphId& the_edgeid,
-        float the_source, float the_target,
-        const sif::Cost& the_cost, float the_turn_cost, float the_sortcost,
-        uint32_t the_predecessor, const baldr::DirectedEdge* the_edge,
-        sif::TravelMode the_travelmode,
-        std::shared_ptr<const sif::EdgeLabel> the_edgelabel)
-      : Label(the_nodeid, kInvalidDestination, the_edgeid,
-              the_source, the_target, the_cost, the_turn_cost, the_sortcost,
-              the_predecessor, the_edge, the_travelmode, the_edgelabel) {}
+class Label : public sif::EdgeLabel {
+ public:
 
-  /**
-   * Construct a Label without a node Id.
-   */
-  Label(uint16_t the_dest, const baldr::GraphId& the_edgeid,
-        float the_source, float the_target,
-        const sif::Cost& the_cost, float the_turn_cost, float the_sortcost,
-        uint32_t the_predecessor, const baldr::DirectedEdge* the_edge,
-        sif::TravelMode the_travelmode,
-        std::shared_ptr<const sif::EdgeLabel> the_edgelabel)
-      : Label({}, the_dest, the_edgeid, the_source, the_target,
-              the_cost, the_turn_cost, the_sortcost,
-              the_predecessor, the_edge, the_travelmode, the_edgelabel) {}
+  Label() {
+    // zero out the data but set the node Id and edge Id to invalid
+    memset(this, 0, sizeof(Label));
+    nodeid_ = baldr::GraphId();
+    edgeid_ = baldr::GraphId();
```

Comment:
I believe this constructor is required since find_shortest_path has a Label label (so the lambda can capture the entire Label - since it needs the entire EdgeLabel to do costing/Allowed methods)